test/extra/PublisherSubscriberTest.cpp

```diff
@@ -87,27 +99,165 @@ TEST_F(PublisherSubscriberTest, SinglePubSingleSub) {
+			message << "Message number: " << remaining;
+
+			auto result =
+			    pub.publish({std::move(message).str(),
+			                 v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT});
+			EXPECT_EQ(result.code(), v1::UCode::OK);
+		}
+	}
+
+	ValidateMessages(rx_queue, num_publish_messages, "Message number: ");
+}
+
+// Single publisher, multiple subscribers (2) on the same topic.
+TEST_F(PublisherSubscriberTest, SinglePubMultipleSub) {
+	auto transport = getTransport();
+
+	communication::Publisher pub(transport, makeUUri(TOPIC_URI),
+	                             v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT);
 
+	// First subscriber and queue
+	std::mutex rx_queue_mtx;
+	std::queue<v1::UMessage> rx_queue;
+	auto on_rx = [&rx_queue_mtx, &rx_queue](const v1::UMessage& message) {
+		std::lock_guard lock(rx_queue_mtx);
+		rx_queue.push(message);
+	};
+	auto maybe_sub = communication::Subscriber::subscribe(
+	    transport, makeUUri(TOPIC_URI), std::move(on_rx));
 	EXPECT_TRUE(maybe_sub);
-	if (!maybe_sub) {
-		return;
+
+	// Second subscriber and queue
+	std::mutex rx_queue_mtx2;
+	std::queue<v1::UMessage> rx_queue2;
+	auto on_rx2 = [&rx_queue_mtx2, &rx_queue2](const v1::UMessage& message) {
+		std::lock_guard lock(rx_queue_mtx2);
+		rx_queue2.push(message);
+	};
+	auto maybe_sub2 = communication::Subscriber::subscribe(
+	    transport, makeUUri(TOPIC_URI), std::move(on_rx2));
+	EXPECT_TRUE(maybe_sub2);
+
+	if (maybe_sub && maybe_sub2) {
+		for (auto remaining = num_publish_messages; remaining > 0;
+		     --remaining) {
+			std::ostringstream message;
+			message << "Message number: " << remaining;
+
+			auto result =
+			    pub.publish({std::move(message).str(),
+			                 v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT});
+			EXPECT_EQ(result.code(), v1::UCode::OK);
+		}
 	}
-	auto sub = std::move(maybe_sub).value();
 
-	constexpr size_t num_publish_messages = 25;
-	for (auto remaining = num_publish_messages; remaining > 0; --remaining) {
-		std::ostringstream message;
-		message << "Message number: " << remaining;
+	ValidateMessages(rx_queue, num_publish_messages, "Message number: ");
+	ValidateMessages(rx_queue2, num_publish_messages, "Message number: ");
+}
+
+// Single publisher, two subscribers on different topics
+TEST_F(PublisherSubscriberTest, SinglePubMultipleSubDifferentTopics) {
+	auto transport = getTransport();
+
+	communication::Publisher pub(transport, makeUUri(TOPIC_URI),
+	                             v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT);
+
+	std::mutex rx_queue_mtx;
+	std::queue<v1::UMessage> rx_queue;
+	auto on_rx = [&rx_queue_mtx, &rx_queue](const v1::UMessage& message) {
+		std::lock_guard lock(rx_queue_mtx);
+		rx_queue.push(message);
+	};
+
+	auto maybe_sub = communication::Subscriber::subscribe(
+	    transport, makeUUri(TOPIC_URI), std::move(on_rx));
+	EXPECT_TRUE(maybe_sub);
+
+	// subscribe to a different topic (non-existent topic)
+	auto on_rx2 = [](const v1::UMessage& message) { FAIL(); };
```

Comment:
I was about to say something about thread safety and gtest, but it looks like gtest _[should be thread safe](https://github.com/google/googletest/tree/main/googletest#multi-threaded-tests)_.